src/lightning/fabric/strategies/xla_fsdp.py

```diff
@@ -0,0 +1,404 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+from contextlib import contextmanager
+from pathlib import Path
+from typing import Any, Callable, Dict, Generator, List, Literal, Optional, Tuple, Union
+
+import torch
+from torch import Tensor
+from torch.nn import Module
+from torch.optim import Optimizer
+
+from lightning.fabric.accelerators import Accelerator
+from lightning.fabric.plugins.io.checkpoint_io import CheckpointIO
+from lightning.fabric.plugins.precision import Precision
+from lightning.fabric.strategies import XLAStrategy
+from lightning.fabric.strategies.fsdp import _apply_filter
+from lightning.fabric.strategies.strategy import _BackwardSyncControl, _validate_keys_for_strict_loading
+from lightning.fabric.utilities.imports import _TORCH_GREATER_EQUAL_2_0
+from lightning.fabric.utilities.rank_zero import rank_zero_only, rank_zero_warn
+from lightning.fabric.utilities.types import _PATH, Optimizable
+
+
+class XLAFSDPStrategy(XLAStrategy):
+    """Strategy for training multiple TPU devices using the
+    :func:`torch_xla.distributed.xla_fully_sharded_data_parallel.XlaFullyShardedDataParallel` method.
+
+    For more information check out https://github.com/pytorch/xla/blob/master/docs/fsdp.md
+    """
+
+    def __init__(
+        self,
+        accelerator: Optional[Accelerator] = None,
+        parallel_devices: Optional[List[torch.device]] = None,
+        checkpoint_io: Optional[CheckpointIO] = None,
+        precision: Optional[Precision] = None,
+        state_dict_type: Literal["full", "sharded"] = "sharded",
+        sync_module_states: bool = True,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(
+            accelerator=accelerator,
+            parallel_devices=parallel_devices,
+            checkpoint_io=checkpoint_io,
+            precision=precision,
+        )
+        self._backward_sync_control = _XLAFSDPBackwardSyncControl()
+
+        self._fsdp_kwargs = kwargs
+        self._state_dict_type = state_dict_type
+        self._sync_module_states = sync_module_states
+
+    def setup_environment(self) -> None:
+        from torch_xla.experimental.pjrt import using_pjrt
+
+        assert self.parallel_devices is not None
+        if using_pjrt() and len(self.parallel_devices) == 1:
+            # spawning only 1 device with PjRT is not supported:
+            # https://github.com/Lightning-AI/lightning/pull/17408#discussion_r1170671732
+            raise NotImplementedError(
+                "The `XLAFSDPStrategy` does not support running on a single device with the PjRT runtime."
+                " Try using all devices or the `SingleTPUStrategy` strategy"
+            )
+
+        self._launched = True
+        rank_zero_only.rank = self.global_rank
+        super().setup_environment()
```

Comment:
I suggest removing this method override, and in the parent using `type(self).__name__` in the error message